### PR TITLE
Added Remote Webcam

### PR DESCRIPTION
**Pull request by meltingice1337**
I added remote webcam using AForge, it may be a little hackish but it
works.

### CODE REVIEW

**Comment by MaxXor** [comment]
Thanks for your contribution. I'll check the code over the next few days and if everything's alright I'll merge it.

**Comment by meltingice1337** [comment]
I see now that i removed something from the main form. Should i just commit the changes to this branch ?

**Comment by MaxXor** [comment]
Yep, you can continue to commit on this branc.

**Comment by meltingice1337** [comment]
I would like to add an icon to TCP Connections in the contextual menu. Should i make a new branch for that ?

**Comment by MaxXor** [comment]
Yes.

**Comment by DragonzMaster** [comment]
Please go back to this issue #424 to know which files are important and remove the the other files, you will see files with .cs0 extension they are not important you can simply remove them from your implementation to reduce size and clean code. :smile:

**Comment by meltingice1337** [comment]
@DragonzMaster I've removed the unused files and the client build was 10kb smaller. Thank you for your input.

File: Client/Client.csproj [hunk 1]
```diff
@@ -66,38 +65,26 @@
...
     <Compile Include="Core\AForge\Video.DirectShow\Internals\IFilterGraph.cs" />
     <Compile Include="Core\AForge\Video.DirectShow\Internals\IFilterGraph2.cs" />
     <Compile Include="Core\AForge\Video.DirectShow\Internals\IGraphBuilder.cs" />
     <Compile Include="Core\AForge\Video.DirectShow\Internals\IMediaControl.cs" />
     <Compile Include="Core\AForge\Video.DirectShow\Internals\IMediaEventEx.cs" />
-    <Compile Include="Core\AForge\Video.DirectShow\Internals\IMediaFilter.cs" />
-    <Compile Include="Core\AForge\Video.DirectShow\Internals\IPersist.cs" />
     <Compile Include="Core\AForge\Video.DirectShow\Internals\IPin.cs" />
     <Compile Include="Core\AForge\Video.DirectShow\Internals\IPropertyBag.cs" />
     <Compile Include="Core\AForge\Video.DirectShow\Internals\IReferenceClock.cs" />
```
Comment (DragonzMaster) [inline Client/Client.csproj]: You can remove this file too ;)

Comment (meltingice1337) [inline Client/Client.csproj]: @DragonzMaster I really don't understand what do you mean. What files should I delete ?

Comment (DragonzMaster) [inline Client/Client.csproj]: This one, IReferenceClock.cs

**Comment by MaxXor** [comment]
Can you post a screenshot of it capturing your webcam?

**Comment by Vigorian** [comment]
for me webcam works fine http://imgur.com/M5XPcm7

**Comment by meltingice1337** [comment]
Here you go http://i.imgur.com/holT4lL.png

**Comment by MaxXor** [comment]
Thanks! Made some coding style adjustments and fixed some bugs. For example when you just open remote webcam and close it, client throws an exception and reconnects (cause `FinalVideo` is null). And you removed the selectAll context menu item by accident. Please take care of things like that next time. Nonetheless good job and thanks for your contribution!

**Pull request merged**